tests/TestCase/ORM/EntityTest.php

```diff
@@ -1060,6 +1103,29 @@ public function testToArrayVirtualProperties()
     }
 
     /**
+     * Tests setting virtual properties with merging.
+     *
+     * @return void
+     */
+    public function testSetVirtualWithMerge()
+    {
+        $data = ['secret' => 'sauce', 'name' => 'mark', 'id' => 1];
+        $entity = new Entity($data);
+        $entity->setVirtual(['secret']);
+
+        $result = $entity->getVirtual();
+        $this->assertSame(['secret'], $result);
+
+        $entity->setVirtual(['name'], true);
+
+        $result = $entity->getVirtual();
+        $this->assertSame(['secret', 'name'], $result);
+
+        $entity->setVirtual(['name'], true);
+        $this->assertSame(['secret', 'name'], $result);
```

Comment:
duplicate assert
and above that  `$entity->setVirtual(['name'], true);`  not affect